packages/template-retail-react-app/app/components/product-tile/index.test.js

```diff
@@ -51,42 +51,71 @@ test('Remove from wishlist cannot be muti-clicked', () => {
     expect(onClick).toHaveBeenCalledTimes(1)
 })
 
-test('Renders variation selection swatches', () => {
+test('Renders variant details based on the selected swatch', async () => {
     const {getAllByRole, getByTestId} = renderWithProviders(
         <ProductTile product={mockProductSearchItem} />
     )
     const swatches = getAllByRole('radio')
     const productImage = getByTestId('product-tile-image')
+    let productTile = getByTestId('product-tile')
 
     // Initial render will show swatched and the image will be the represented product variation
     expect(swatches).toHaveLength(2)
     expect(productImage.firstChild.getAttribute('src')).toBe(
         'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw175c1a89/images/large/PG.33698RUBN4Q.CHARCWL.PZ.jpg'
     )
+    let currentPriceTag = productTile.querySelectorAll('b')
+    let strikethroughPriceTag = productTile.querySelectorAll('s')
+    expect(currentPriceTag).toHaveLength(1)
+    expect(within(currentPriceTag[0]).getByText(/£191\.99/i)).toBeDefined()
+    expect(strikethroughPriceTag).toHaveLength(1)
+    expect(within(strikethroughPriceTag[0]).getByText(/£320\.00/i)).toBeDefined()
 
-    // Hovering over color swatch changes the image.
-    fireEvent.mouseEnter(swatches[1])
-    waitFor(() => {
-        expect(productImage.firstChild.getAttribute('src')).toBe(
-            'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw29b7f226/images/large/PG.52002RUBN4Q.NAVYWL.PZ.jpg'
-        )
-    })
+    // Navigating to different color swatch changes the image & price.
+    const swatchGroup = screen.getByRole('radiogroup').parentNode
+    fireEvent.keyDown(swatchGroup, {key: 'ArrowLeft', code: 'ArrowRight', charCode: 37})
+    await waitFor(() => screen.getByTestId('product-tile-image'))
+    expect(screen.getByTestId('product-tile-image').firstChild.getAttribute('src')).toBe(
+        'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw29b7f226/images/large/PG.52002RUBN4Q.NAVYWL.PZ.jpg'
+    )
+    productTile = screen.getByTestId('product-tile')
```

Comment:
Correct me if I am wrong, It is not likely we have to re-query here since the testing library is keeping track on the value of the nodes. See [here](https://testing-library.com/docs/react-testing-library/faq/).  🤔

Reply:
I think you are right. I added while debugging some of the async issues. I will fix it.